cpp/include/Ice/ProxyFunctions.h

```diff
@@ -31,6 +32,21 @@ void checkNotNull(std::optional<Prx> prx, const Current& current)
     }
 }
 
+/**
+ * Verifies that a proxy is not null, and throws a MarshalException if it is.
+ * @param prx The proxy to check.
+ * @param message The message used to initialize the MarshalException.
+ * @throw MarshalException If the proxy is null.
+ * */
+template<typename Prx, std::enable_if_t<std::is_base_of<ObjectPrx, Prx>::value, bool> = true>
+void checkNotNull(std::optional<Prx> prx, const std::string& message)
```

Comment:
From an API point of view, should the message be:
- a string_view, because the typical usage is to initialize it with a string literal?
- a string because the typical usage is to create the string (as in `"operation " + opName + " provided  a null proxy"`) and MarshalException then adopts this string.

Since it's presumably rare (and the check should be inexpensive), string_view sounds more correct.

Reply:
After looing at the actual usage, I think the API is not correct. We should create the error message only when prx is null. The simplest is to not use a helper and just write inline:
```cpp
if (!prx)
{
   // manufacture error message with an ostringstream ...
   throw MarshalException{__FILE, __LINE__, message};
}
```

Reply:
Fixed as suggested